macros/models/_project/chainlink/chainlink_ccip_macro.sql

```diff
@@ -0,0 +1,46 @@
+{% macro 
+    chainlink_ccip_tokens_transferred(
+        blockchain
+    ) 
+%}
+
+SELECT DISTINCT
+    le.blockchain,
+    le.block_time,
+    le.tx_hash,
+    COALESCE(le.total_tokens / POWER(
+        CAST(10 AS double precision),
+        CAST(t.decimals AS double precision)
+    ), 0) AS total_tokens,
+    COALESCE(sr.destination_blockchain, 'UNKNOWN') AS destination_blockchain,
+    te.contract_address AS token,
+    COALESCE(t.symbol, 'UNKNOWN') AS token_symbol,
+    COALESCE(p.price * (
+        le.total_tokens / POWER(
+            CAST(10 AS double precision),
+            CAST(t.decimals AS double precision)
+        )
+    ), 0) AS token_price
+FROM
+    {{ref('chainlink_' ~ blockchain ~ '_ccip_tokens_transferred_logs')}} le
+    INNER JOIN {{ref('chainlink_' ~ blockchain ~ '_ccip_send_requested')}} sr ON le.tx_hash = sr.tx_hash
```

Comment:
are there more join conditions we can add here to improve performance?
`block_number`? `block_date`? `block_time`? `evt_index`?

Reply:
joining only on varbinary hash values is a bit slow